### ISSUE TITLE
Code that moves and cleans up objects as they move from /deposit to storage roots doesn't always work as expected

### ISSUE THREAD

**Issue by julianmorley** [sul-dlss/preservation_catalog#1667]
**Describe the bug**
The problem documented in https://github.com/sul-dlss/preservation_catalog/issues/1633 still occurs in production.

**User Impact**
Accessioning experiences issues (delays, failed accessioning, other errors), especially during periods of heavy activity.

**Expected behavior**
Files in /deposit should be consistently moved into pres storage roots without errors, and any artifacts left in /deposit should be consistently cleaned up.

**Additional context**
This issue is caused by how Ceph handles the creation and deletion of hard links; these actions are asynchronous and during periods of heavy activity may take seconds to complete. So if the code is trying to remove the object from /deposit, thinking that it has been successfully moved to the storage root, but the object has not yet finished creating the new links, a read/permissions occur will occur. Sometimes this is resolved by the passage of time, but a more consistent fix is to restart the metadata server managing that storage root.

It looks like this:
```
ls: cannot access /pres-01/sdr2objects/qt/116/cp/8268/qt116cp8268/v0004/manifests/versionAdditions.xml: Permission denied
total 22
-rw-rw-r-- 1 pres pres 6733 Jan 24 09:28 fileInventoryDifference.xml
-rw-rw-r-- 1 pres pres 1606 Jan 24 09:28 manifestInventory.xml
-rw-rw-r-- 1 pres pres 7971 Jan 24 09:28 signatureCatalog.xml
-????????? ? ?    ?       ?            ? versionAdditions.xml
-rw-rw-r-- 1 pres pres 4634 Jan 24 09:28 versionInventory.xml
```

**Comment by julianmorley** [comment]
The actual fix is probably in preservation_robots; see https://github.com/sul-dlss/preservation_robots/issues/277 for context. Some additional logic or checks or re-ordering of actions in pres robots would probably address this. Maybe something like 
* create hardlinks in destination storage root, 
* check that all are readable in the destination (and keep trying until they arre)
* remove hardlinks from source, check that they're all removed
* *then* report that the object has been accessioned to preservation catalog for zipmaker etc to do work

**Comment by jmartin-sul** [comment]
i would agree that for this `/deposit` related issue, the fix should very likely be in pres robots.

should we transfer this issue to that repo?

cc: @andrewjbtw

**Comment by julianmorley** [comment]
I'm a-OK to transfer this to pres robots. 

Another random-ish bit of data; versionAdditions.xml is often (but not always) the only file impacted by this. Looking at the pres linker code, I think it's just because it tends to be the last file modified and so is more likely to not be done with re-linking by the time the next step tries to process that directory.
